csi/controller_test.go

```diff
@@ -3371,3 +3376,162 @@ func TestGetCapacity(t *testing.T) {
 	assert.NotNil(t, res)
 	assert.Equal(t, int64(0), res.AvailableCapacity)
 }
+
+func TestOsdCsiServer_CreateSnapshot(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockCloudBackupClient := mock.NewMockOpenStorageCloudBackupClient(ctrl)
+
+	ctx := context.Background()
+
+	mockErr := errors.New("MOCK ERROR")
+	creationTime := timestamppb.Now()
+
+	mockCloudBackupClient.EXPECT().Create(gomock.Any(), gomock.Any(), gomock.Any()).
```

Comment:
in our call, you mentioned this needs to be a part of the Csi struct in order for this to work.

However, we'll need to find a way to generate this client per test. The reason is that we need a new client for each remote conn we get for round-robin connections.